### DISCUSSION TITLE
ORM Structured response

### DISCUSSION THREAD

**Discussion by Saiprasadlokhande** [serverpod/serverpod#617] [Q&A] [answered]
Do Serverpod ORM have a structured response, as I can see we have been returning, bool/Class/List<Class>. 
Can we enhance our ORM response to a Default response class, where we can actually send statusCode, body, and message as well, to handle most of the error messages directly from server.

**Comment by vlidholt** [comment, marked as answer]
Hi! You typically want to include the response information in your returned serializable objects, rather than passing a http status code. Serverpod itself will set the status code depending on the call itself being successful, if an exception is thrown on the server, or if access is denied.

Very soon, it will also be possible to cast an error on the server and catch it on the client.

**Reply by fischerscode** [reply]
Thought about this, too. I think it would be great if one would be able to throw a special class, that takes the response code (not 200) and a error message. Serverpod (server) would catch the throw, detect the special type and send it to the client. If a different class was thrown, Serverpod should default to 500 - `Internal server error.`.
At the client side, 'deserialization' would be easy, since the response code determines the type (response or error).
